src/hooks/useAppState/middleware.js

```diff
@@ -16,34 +18,44 @@ const applyMiddleware = (dispatch, getState) => (action) => {
   switch (type) {
     case types.DELETE_ITEM_BY_KEY: {
       const state = getState();
-      const itemToRemove = state.items.find((item) => item.key === data.key);
+      const itemToRemove = selectItems(state).find((item) => item.key === data.key);
       if (itemToRemove) {
-        deleteItem(state.listId, itemToRemove.id);
+        deleteItem(selectListId(state), itemToRemove.id);
       }
       break;
     }
 
     case types.UPDATE_ITEM_BY_KEY: {
       const state = getState();
-      const itemToUpdate = state.items.find((item) => item.key === data.key);
+      const itemToUpdate = selectItems(state).find((item) => item.key === data.key);
       if (itemToUpdate) {
-        updateItem(state.listId, itemToUpdate.id, data.text);
+        updateItem(selectListId(state), itemToUpdate.id, data.text);
       }
       break;
     }
 
     case types.NEW_ITEM: {
       const { key, text } = data;
       const state = getState();
-      createItem(state.listId, text)
+      createItem(selectListId(state), text)
         .then((id) => dispatch({ type: types.STORE_ID, data: { key, id } }));
       break;
     }
 
-    case types.SET_LIST_ID: {
-      if (!data.listId) return;
-      getItems(data.listId)
-        .then((items) => dispatch({ type: types.MERGE_ITEMS, data: { items } }));
+    case types.SET_STATE: {
+      const lists = data.state?.lists ?? [];
+      if (lists?.length === 0) {
+        const name = 'Shopping List';
+        const key = `${Date.now()}`;
+        dispatch({ type: types.NEW_LIST, data: { key, name } });
+        createList(name)
+          .then((id) => dispatch({ type: types.STORE_LIST_ID, data: { key, id } }));
+      } else {
+        lists.forEach(({ id }) => {
+          getItems(id)
+            .then((items) => dispatch({ type: types.MERGE_ITEMS, data: { items, listId: id } }));
+        });
+      }
```

Comment:
When we first set the state (when the app is first loaded) if there is no list ID in the state we created one. If there is one or more list IDs we fetch all the items from it and merge them with the items we have locally.